python/whylogs/experimental/extras/embedding_metric.py

```diff
@@ -128,16 +129,27 @@ def to_protobuf(self) -> MetricMessage:
             msg["references"] = self.references.to_protobuf()
 
         return MetricMessage(metric_components=msg)
-
     def _update_submetrics(self, submetric: str, data: PreprocessedColumn) -> None:
         for key in self.submetrics[submetric].keys():
             self.submetrics[submetric][key].columnar_update(data)
 
     def columnar_update(self, data: PreprocessedColumn) -> OperationResult:
-        if data.numpy.ints is None and data.numpy.floats is None:
+        if data.pandas.tensors is None and data.list.tensors is None:
             return OperationResult.ok(0)
 
-        matrices = [X for X in [data.numpy.ints, data.numpy.floats] if X is not None]
+        pandas_tensors = None
+        if data.pandas.tensors is not None:
```

Comment:
Tensorizing should happen in `PreprocessedColumn` so we don't have to repeat this code in every tensorish metric.

Reply:
Yes, I'm on board for this now.